apps/src/templates/levelSummary/SummaryContainer.jsx

```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types';
+import React, {useState, useEffect} from 'react';
+import {Provider} from 'react-redux';
+
+import InstructorsOnly from '@cdo/apps/code-studio/components/InstructorsOnly';
+import QuestionRenderer from '@cdo/apps/templates/levelSummary/QuestionRenderer';
+import SubLevelDropdown from '@cdo/apps/templates/levelSummary/SubLevelDropdown';
+import SummaryResponses from '@cdo/apps/templates/levelSummary/SummaryResponses';
+import SummaryTeacherInstructions from '@cdo/apps/templates/levelSummary/SummaryTeacherInstructions';
+
+import styles from './summary.module.scss';
+
+const SummaryContainer = ({store, scriptData, isLevelGroup}) => {
+  const [updatedScriptData, setUpdatedScriptData] = useState({
+    ...scriptData,
+    responses: scriptData.responses[0],
+  });
+  const [subLevelNum, setSubLevelNum] = useState(0);
+
+  useEffect(() => {
+    const levelData = scriptData.levels[subLevelNum];
+    const levelResponses = scriptData.responses[subLevelNum];
+
+    setUpdatedScriptData({
+      ...scriptData,
+      viewing_level_data: levelData,
+      responses: levelResponses,
+    });
+  }, [subLevelNum, scriptData]);
```

Comment:
```suggestion
  const updatedScriptData = useMemo(() => {
    const levelData = scriptData.levels[subLevelNum];
    const levelResponses = scriptData.responses[subLevelNum];

    return({
      ...scriptData,
      viewing_level_data: levelData,
      responses: levelResponses,
    });
  }, [subLevelNum, scriptData]);
```
useState is usually best for small pieces of data that are updated by a user action like a button click. Because this is only updated by useEffect, I think it would be much easier to use `useMemo`

Reply:
I think it would also be nice to have three variables `currentLevel`, `currentResponses` and `scriptData` that we pass around instead of one variable `updatedScriptData` that we pass around and modify.
1) We could then determine which level is being viewed (the currentLevel) on the FE instead of adding `viewing_level_data` to the api response
2) It is closer to best practices because generally we don't want to modify the api response/scriptdata after it has been retrieved

Reply:
I refactored it to just use `scriptData` and `subLevelNum`! Thanks for this suggestion! Let me know what you think about my solution.